### ISSUE TITLE
How to add the Metrics into Manuscripts

### ISSUE THREAD

**Issue by aswanipranjal** [chaoss/grimoirelab-manuscripts#50]
As per the discussion with Jesus, this ticket is about how to add code for Metrics into Manuscripts.

- We can start with writing dirty code in Jupyter-notebooks and structuring the code there and then once it gets finalised, we can add clean code into the Master branch so that the commit history has only clean code and definite updates on the project.

- When we want to add a functionality into Manuscripts, a new issue can be created and that can be linked back to this main issue.

Other ideas are welcome! :D

**Comment by jgbarah** [comment]
In parallel to work with the notebooks, I think we should start studying which changes Manuscripts would need, and the schema we would use in it. For now, I think we could design a report which, if the `--chaoss` (for example) option is specified, a specific section "CHAOSS Metrics" is produced, with all the GMD metrics that can be computed for the specified data sources.

If you agree with this simple specification, we could start looking at the code that needs to be changed, and how to change it.

**Comment by aswanipranjal** [comment]
@jgbarah, I am a little confused. Why do we need to have a `--chaoss` argument? Aren't GMD metrics already a part of CHAOSS? 

I was thinking of dividing the report into 4 parts, each one corresponding to Diversity-Inclusion, GMD, Risk and Value.

Can you please expand on the `--chaoss` argument part?
I am sorry if I am missing something very obvious here.

**Comment by jgbarah** [comment]
Currently, Manuscripts has some functionality (produces a certain kind of report). The idea of the --chaoss option is for including, optionally, in addition to that functionality, the CHAOSS metrics. Which for now will be the GMD metrics.

The main reason is that Manuscripts already has users, so that I would prefer not to break current functionality (the kind of report it produces now). If users don't include `--chaoss`, it will generate the current report, and that's it. But if they include the `--chaoss`flag, the new section(s) with CHAOSS metrics will be included as well.

Is it more clear now?

**Comment by aswanipranjal** [comment]
@jgbarah, I've made some changes to manuscripts, specifically `metrics.py` and `esquery.py` to be able to realise the mertics. I was thinking that I can create another branch for now in my forked version of manuscripts for testing and development purposes and then you (or anyone who wants to collaborate on manuscripts) can download the [repository](https://github.com/aswanipranjal/gsoc-manuscripts) (gsoc version of manuscripts) and initialise the submodule inside it and use the Notebooks. That way, when we have finalised everything, we can proceed to merge parts of it into the master branch of chaoss/grimoirelab-manuscripts.

What do you think?
